### PR TITLE
port cli/features/tutorial.feature to Go

### LINKED ISSUE #369: port cli/features/tutorial.feature to go
After moving this feature into exo-go, I believe we can remove the cli folder

### PR DESCRIPTION
**Pull request by cheahuychou**
resolves #369

### CODE REVIEW

File: exo-go/features/tutorial.feature [hunk 1]
```diff
@@ -117,7 +117,7 @@ Feature: Following the tutorial
 
 
   # Scenario: starting the application
-  #   When starting "exo run" in this application's directory
+  #   When starting "exo run" in my application directory
```
Comment (charlierudolph) [inline exo-go/features/tutorial.feature]: Please remove this commented out scenario.

File: exo-go/test_helpers/index.go [hunk 2]
```diff
@@ -33,6 +33,18 @@ func CheckoutApp(cwd, appName string) error {
...
+func checkoutServiceTemplate(appDir, templateName string) error {
+	_, filePath, _, _ := runtime.Caller(0)
+	src := path.Join(path.Dir(filePath), "..", "..", "exosphere-shared", "templates", "boilr-templates", templateName)
+	dest := path.Join(appDir, ".exosphere", templateName)
+	err := os.RemoveAll(dest)
+	if err != nil {
+		return err
+	}
+	return CopyDir(src, dest)
+
```
Comment (charlierudolph) [inline exo-go/test_helpers/index.go]: Please remove this extra newline

File: exo-go/test_helpers/tutorial_feature_context.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		formattedText, err := html2text.FromString(string(body))
+		if err != nil {
+			return err
+		}
+		text := util.Strip(`\*+\n|\-+\n`, formattedText)
```
Comment (charlierudolph) [inline exo-go/test_helpers/tutorial_feature_context.go]: Can you please add a comment as to what exactly is being stripped out here?

Comment (cheahuychou) [reply exo-go/test_helpers/tutorial_feature_context.go]: This strips out the `*********` and `----------` lines that `hmtl2text` adds to the text https://github.com/jaytaylor/html2text#example-usage

**Comment by cheahuychou** [comment]
@charlierudolph comments addressed

**Review by charlierudolph — APPROVED** [review]

File: exo-go/test_helpers/tutorial_feature_context.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		return nil
+	})
+
+	s.Step(`^my application contains the template folder "([^"]*)"$`, func(templateDir string) error {
+		templateName := strings.Split(templateDir, "/")[1]
+		return checkoutServiceTemplate(appDir, templateName)
+	})
+
+	s.Step(`^waiting until I see "([^"]*)" in the terminal$`, func(expectedText string) error {
+		return childCmdPlus.WaitForText(expectedText, 60000)
```
Comment (charlierudolph) [inline exo-go/test_helpers/tutorial_feature_context.go]: 60000 -> `time.Minute`

File: exo-go/test_helpers/tutorial_feature_context.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/Originate/exosphere/exo-go/src/util"
+	"github.com/jaytaylor/html2text"
+)
+
+// TutorialFeatureContext defines the festure context for the tutorial
+// nolint gocyclo
+func TutorialFeatureContext(s *godog.Suite) {
+
+	s.Step(`^I see "([^"]*)" in the terminal$`, func(expectedText string) error {
+		return childCmdPlus.WaitForText(expectedText, 1000)
```
Comment (charlierudolph) [inline exo-go/test_helpers/tutorial_feature_context.go]: 1000 -> `time.Second`

**Review by charlierudolph — APPROVED** [review]

**Pull request closed without merging**